src/cmake/thirdparty/FindMFEM.cmake

```diff
@@ -31,36 +31,97 @@ if(_use_mfem_config)
               ${MFEM_DIR}/share/mfem
               ${MFEM_DIR}/mfem)
 
-    if(NOT MEFM_LIBRARY)
-       set(MFEM_LIBRARY ${MFEM_LIBRARIES})
+else()
+
+    find_path(
+        MFEM_INCLUDE_DIRS mfem.hpp
+        PATHS ${MFEM_DIR}/include
+        NO_DEFAULT_PATH
+        NO_CMAKE_ENVIRONMENT_PATH
+        NO_CMAKE_PATH
+        NO_SYSTEM_ENVIRONMENT_PATH
+        NO_CMAKE_SYSTEM_PATH
+    )
+
+    find_library(
+        MFEM_LIBRARIES NAMES mfem
+        PATHS ${MFEM_DIR}/lib
+        NO_DEFAULT_PATH
+        NO_CMAKE_ENVIRONMENT_PATH
+        NO_CMAKE_PATH
+        NO_SYSTEM_ENVIRONMENT_PATH
+        NO_CMAKE_SYSTEM_PATH )
+
+
+    # when MFEM is built w/o cmake, we can get the details
+    # of deps from its config.mk file
+    find_path(
+        MFEM_CFG_DIR config.mk
+        PATHS ${MFEM_DIR}/share/mfem/
+        NO_DEFAULT_PATH
+        NO_CMAKE_ENVIRONMENT_PATH
+        NO_CMAKE_PATH
+        NO_SYSTEM_ENVIRONMENT_PATH
+        NO_CMAKE_SYSTEM_PATH
+    )
+
+    if(NOT MFEM_CFG_DIR)
+        message(FATAL_ERROR "Failed to find any MFEM build configuration files in ${MFEM_DIR}")
+    else()
+        message(STATUS "Using MFEM's GNU Make config file: ${MFEM_CFG_DIR}/config.mk")
     endif()
 
-else()
+    # read config.mk file
+    file(READ "${MFEM_CFG_DIR}/config.mk" mfem_cfg_file_txt)
```

Comment:
Yikes! Is this the preferred way to do this?

Reply:
I don't know about "preferred" but using their info makefile config in CMake is less than ideal.